src/components/CustomerDetails/CollectCustomerDetailsScreen.tsx

```diff
@@ -172,7 +172,35 @@ const CollectCustomerDetailsScreen: FunctionComponent<NavigationFocusInjectedPro
     );
   }, [selectionArray, selectedIdType, setSelectedIdType]);
 
+  const setState = useState()[1];
   const onCheck = async (input: string): Promise<void> => {
+    /**
+     * If "Passport" tab is chosen, we want to extract the passport
+     * number from a JSON object containing the `passportId`.
+     */
+    if (selectedIdType.label === "Passport") {
+      let passportId;
+      try {
+        ({ passportId } = JSON.parse(input));
+      } catch (e) {
+        /**
+         * If we encounter any JSON errors, we set `passportId` to empty
+         * in order to allow logic below to handle it as an empty/invalid ID.
+         *
+         * However, any other errors will trigger the ErrorBoundary.
+         */
+        if (e instanceof SyntaxError) {
+          passportId = "";
+        } else {
+          setState(() => {
```

Comment:
can't seem to find the answer on google 😢 .
What `state` is this update function changing? Since you only retrieve `useState()[1]` in line 175

Reply:
Since we want the error to be thrown to the error boundary instead, we'll have to update React this way since React error boundaries don't actually catch errors in async code. You can see [this](https://reactjs.org/docs/error-boundaries.html)!